src/main/java/DianaAssistant.java

```diff
@@ -0,0 +1,155 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class DianaAssistant {
+    private static final String ENCLOSURE = "------------------------------";
+
+    public void interact() {
+        printWelcomeMessage();
+        Scanner scanner = new Scanner(System.in);
+        List<Task> list = new ArrayList<>();
+
+        String input;
+        while (true) {
+            input = scanner.nextLine();
+            if ("bye".equals(input)) {
+                break;
+            }
+
+            switch(getCommand(input)) {
+                case "list":
+                    printList(list);
+                    break;
+                case "mark":
+                    toMark(list, input, true);
+                    break;
+                case "unmark":
+                    toMark(list, input, false);
+                    break;
+                case "todo":
+                    addTodo(input, list);
+                    break;
+                case "event":
+                    addEvent(input, list);
+                    break;
+                case "deadline":
+                    addDeadline(input, list);
+                    break;
+                default:
+                    System.out.println("Unknown command");
+                    break;
+            }
+        }
+
+        System.out.println("Bye! Hope to see you again soon.");
+        scanner.close();
+    }
+
+    private String getCommand(String input) {
+        return input.split(" ")[0];
+    }
+
+    private void printWelcomeMessage() {
+        printEnclosure();
+        System.out.println("Hello, I am Diana! A personal assistant that helps you keep track of your tasks :)");
+        System.out.println("Here are the list of things that Diana can do for you");
+        System.out.println("1. Create a todo task eg. [todo read book]");
+        System.out.println("2. Create an event task eg. [event read book /from 2pm /to 4pm]");
+        System.out.println("3. Create a deadline task eg. [deadline read book /by 2pm]");
+        System.out.println("4. Type either mark or unmark to indicate the completion rate of your task");
+        System.out.println("5. Type list to view the your list of tasks.");
+        System.out.println("6. Type bye to exit the programme");
+        printEnclosure();
+    }
+
+    private void printEnclosure() {
+        System.out.println(ENCLOSURE);
+    }
+
+    private void printList (List<Task> list) {
+        printEnclosure();
+
+        for (int i = 0; i < list.size(); i++) {
+            Task task = list.get(i);
+            System.out.println((i + 1) + ". " + task.toString());
+        }
+        printEnclosure();
+        return;
+    }
+
+    private void toMark (List<Task> list, String input, boolean toMark) {
+        try {
+            int TaskNum = Integer.valueOf(input.substring(input.indexOf(" ") + 1)) - 1;
```

Comment:
Could you make this expression simpler by splitting it into three separate steps, namely, getting the index of " ", getting the substring, then converting it to an integer?